### PR TITLE
Minor CI improvements

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR bundles a few minor CI improvements together. It:

- fixes shellcheck warnings in all distribution shell scripts
- fixes a build issue where we were recompiling everything all the time, which caused concurrency issues when running multiple test stages on the same workspace in CI, resulting in missing classes at runtime
- updates `expression-language` module to use the right parent module
- updates `scala-maven-plugin` to compile Java code for the configured version and not whatever the current JVM is (this previously worked since we would recompile the Java sources after anyway)
- skips compilation in non-build stages (e.g. testing, analysis, etc.) to avoid concurrent build issues; unfortunately some things were still being compiled and I couldn't track it down, so right now I explicitly skip compilation in the parallel stages
- replaces deprecated `MaxRAMPercentage` with `MaxRAMFraction`, since they're essentially the same
- runs the randomized tests in parallel to the unit tests

I would recommend reviewing this commit by commit, as one commit for example was just fixing warnings/style issues, no real logic changes.

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by npepinpe** [comment]
@oleschoenburg - as you've already had a bit of insight in the CI, I figured we might continue a little :slightly_smiling_face: Let me know however if you don't feel comfortable reviewing this, or want to do it together

File: .ci/scripts/distribution/random-test-java.sh [hunk 1]
```diff
@@ -10,19 +10,20 @@ JUNIT_THREAD_COUNT=${JUNIT_THREAD_COUNT:-}
 MAVEN_PROPERTIES=(
   -DskipITs
   -DskipChecks
-  -DtestMavenId=1
+  -DtestMavenId=4
```
Comment (oleschoenburg) [inline .ci/scripts/distribution/random-test-java.sh:13]: :question:  I'm not sure where this is coming from & what it means. Can you explain? :)

File: expression-language/pom.xml [hunk 2]
```diff
@@ -4,11 +4,11 @@
 
   <parent>
     <groupId>io.camunda</groupId>
-    <artifactId>zeebe-root</artifactId>
+    <artifactId>zeebe-parent</artifactId>
     <version>1.3.0-SNAPSHOT</version>
+    <relativePath>../parent/pom.xml</relativePath>
   </parent>
 
-  <groupId>io.camunda</groupId>
```
Comment (oleschoenburg) [inline expression-language/pom.xml:11]: :question: Is the `groupId` now properly inherited from the parent or why did you remove this here?

File: .ci/scripts/distribution/it-java.sh [hunk 3]
```diff
@@ -11,33 +11,34 @@ MAVEN_PROPERTIES=(
   -DskipUTs
   -DskipChecks
   -DtestMavenId=2
-  -Dfailsafe.rerunFailingTestsCount=7
+  -Dfailsafe.rerunFailingTestsCount=3
```
Comment (oleschoenburg) [inline .ci/scripts/distribution/it-java.sh:14]: :question: Is the goal here that failing tests fail faster? Doesn't this also mean that flaky tests break the pipeline more often?

**Review by lenaschoenburg — COMMENTED** [review]
I like all the shellcheck fixes and having random tests in a separate step totally makes sense. Only a couple of questions from my side, other than that it looks good!

File: .ci/scripts/distribution/random-test-java.sh (hunk 1 above)
Comment (npepinpe) [reply .ci/scripts/distribution/random-test-java.sh:13]: It's used in our `io.camunda.zeebe.test.util.socket.SocketUtil` class. It's used to allocate specific port ranges to each stage/fork to ensure there are no port collisions for tests running in the same pod.

File: expression-language/pom.xml (hunk 2 above)
Comment (npepinpe) [reply expression-language/pom.xml:11]: Exactly :+1: It doesn't hurt to leave, but it is redundant and flagged as a warning.

File: .ci/scripts/distribution/it-java.sh (hunk 3 above)
Comment (npepinpe) [reply .ci/scripts/distribution/it-java.sh:14]: We already fail the build when a test is flaky, but we still rerun it 7 times. This is too high in my opinion, after 3 times, if it's still failed, it's highly likely it's a deterministic failure.

File: .ci/scripts/distribution/random-test-java.sh (hunk 1 above)
Comment (oleschoenburg) [reply .ci/scripts/distribution/random-test-java.sh:13]: Ah, and now we have to increase it because regular tests are running at the same time?

**Review by lenaschoenburg — APPROVED** [review]

**Comment by npepinpe** [comment]
bors merge
